### DISCUSSION TITLE
Support custom parsing of locale url in sub-path routing

### DISCUSSION THREAD

**Discussion by rutkowskib** [vercel/next.js#28245] [Ideas]
### Describe the feature you'd like to request

Currently next.js supports only sub-path locale detection when URL is in format example.com/locale/path. I would like to be able to use URLs in format example.com/country-locale/path where country is locale my app uses to decide which content it should show. 

There has been a few discussions about that ([https://github.com/vercel/next.js/discussions/21871](url), [https://github.com/vercel/next.js/discussions/25215](url)) but none of them is feature request so I post a new one. 

### Describe the solution you'd like

I would like to be able to define in i18n options a parameter which determines how subpath should be parsed e.g 
`i18n: {
    locales: ['en-US', 'fr', 'nl-NL'], 
    defaultLocale: 'en-US',
    subpathFormat: 'country-locale',
  }`

I am willing to help implementing this feature or implement it as a whole.

### Describe alternatives you've considered

I have considered using other strategies such as domain routing and supported form of sub-path localization, but I need other format due to backwards compatibility.

**Comment by ahakem** [comment]
I need the same thing, and currently not using the Link component provided by next, and have a function to structure the correct url

**Comment by kristijorgji** [comment]
I need custom subpaths,
example domain.com/de to use locale de_DE
right now it is like domain.com/de_DE
but I want to be possible to map any custom subpath to a given locale, then can have more user friendly names like
domain.com/de/ or anything instead of de

**Comment by reuixiy** [comment]
Same wishes!

**Comment by bringmetheaugust** [comment]
any updates?

**Comment by eduluq** [comment]
any updates here? thanks!
